### DISCUSSION TITLE
Welcome to BlendArMocap Discussions!

### DISCUSSION THREAD

**Discussion by cgtinker** [cgtinker/BlendArMocap#133] [Announcements]
## 👋 Welcome!
Discussions are used as a place to connect with other users of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Daniel-W-Blender-Python** [comment]
Hello! My name is Daniel, and I have been using blender for a little over two years now. For the past few moths, I've been working on an addon similar to yours. After using BlendAr Mocap, I was inspired to create a similar motion capture addon of my own. It works with mediapipe, and tracks using drivers to transfer animation, but I want to continue to improve it. One of the main issues I'm encountering is that when the face becomes closer or farther from the camera, the drivers in blender expand farther apart from each other, or shrink closer to each other. I am curious to know how you combated this problem in both BlendArMocap and BlendArTrack. While I have been able to limit the effects of this problem by applying simple equations and mathematical remedies, it is still not enough to be usable in my next version. I did want to say that I am very impressed with your work, and I think it is extremely convenient and scholarly. For reference to my project, here is the GitHub page: https://github.com/Daniel-W-Blender-Python/Blender-Motion-Capture-Addon
I would love to hear your feedback on my project, and anything I can change to improve my work. Thank you so much!

**Reply by cgtinker** [reply]
Hey there,

Took a quick look, seems like a fun project!

There are a couple of options:
1. Calculate rotations & distances and use those for mapping [blendarmocap](https://github.com/cgtinker/BlendArMocap/tree/main/src/cgt_core/cgt_calculators_nodes), [rotations](https://github.com/cgtinker/mediapipe_rotations).
2. Calculate the local space of the point cloud - this should be doable with matrices. Mapping from global space won't work.
3. Mediapipe published shape keys for the face, mapping with those will be way easier and may provide better results than the options above.


Best regards and gl!

**Reply by Daniel-W-Blender-Python** [reply]
Thank you so much for your help!
